classes/tdx-signed-fit-image.inc

```diff
@@ -21,3 +21,14 @@ FIT_HASH_ALG ?= "sha256"
 FIT_SIGN_ALG ?= "rsa2048"
 FIT_SIGN_NUMBITS ?= "2048"
 FIT_SIGN_INDIVIDUAL = "0"
+
+# stronger assignments to override settings in meta-ti
+#
+# NOTICE: a side effect of these assignments is that users will need to use
+#         :forcevariable (or an override stronger than :k3) to set them.
+UBOOT_SIGN_KEYDIR:k3 ?= "${TOPDIR}/keys/fit"
+UBOOT_SIGN_KEYNAME:k3 ?= "dev"
+UBOOT_SIGN_IMG_KEYNAME:k3 ?= "dev2"
+FIT_HASH_ALG:k3 ?= "sha256"
+FIT_SIGN_ALG:k3 ?= "rsa2048"
+FIT_SIGN_NUMBITS:k3 ?= "2048"
```

Comment:
Alright. So I had an idea that might work and fix this issue without any side efffect. What if we create a custom variable for each FIT variable that we want to set as default. Something like this:

```
TDX_UBOOT_SIGN_ENABLE ?= "${TOPDIR}/keys/fit"
UBOOT_SIGN_KEYDIR = "${TDX_UBOOT_SIGN_ENABLE}"

TDX_UBOOT_SIGN_KEYNAME ?= "dev"
UBOOT_SIGN_KEYNAME = "${TDX_UBOOT_SIGN_KEYNAME}"

TDX_UBOOT_SIGN_IMG_KEYNAME ?= "dev2"
UBOOT_SIGN_IMG_KEYNAME = "${TDX_UBOOT_SIGN_IMG_KEYNAME}"
```

With this:

- We can have our defaults without worring about being override by a machine.
- The user can easily override it without using forcevariable.
- We don't duplicate the same value indifferent variables (one for iMX and one for K3), making it easy to maintain.

This would also mean we would need to document those variables.

Makes sense?

Reply:
That's a great idea. Thanks! I'll go ahead and do it.

Reply:
@sergioprado When starting to make the changes I thought of an issue that made me stop: if we follow this approach, we'd be overriding settings done by customers at the moment, i.e. it would be a breaking change, right?

Then I further thought of some other points against it:

1. In the end we'd be doing this just to compensate for an issue that only happens when the TI layer is in use; at the moment we have one machine using that layer which means the decision would be based on something (I hope) is an exception case.
2. We don't implement the FIT image signing in our layer, but with the proposed change we'd be adding a group of variables to configure the signing that is provided by another layer; however, we wouldn't override each and every configuration variable which means a user would need to always check both our documentation/code and openembedded's to figure out how to appropriately apply the settings they want. Personally, I'd prefer that users resort mainly to openembedded docs.
3. Though I understand that having our defaults looks good at first sight, I'm not sure if we should override settings from the layer providing the feature unless strictly necessary for our functionalities to work; I say that because I imagine the providers of the feature should only change the defaults when there is a very good reason for it and they should know better than us how to best configure their feature. Of course the same statement holds for meta-ti. A question to comes to mind is: why is our defaults better than the ones from meta-ti and why are the meta-ti ones better than the ones from openembedded?
4. And let me just repeat: this would be a breaking change from NXP users perspective.

Anyways, we should probably talk to avoid a huge thread on this topic :slightly_smiling_face:...

Reply:
Yeah, that is really unfortunate. Alright, to not block this PR, lets leave it as is right now and maybe later we re-evaluate the situation.